snuba/cli/replacer.py

```diff
@@ -33,18 +33,21 @@
 @click.option('--log-level', default=settings.LOG_LEVEL, help='Logging level to use.')
 @click.option('--dogstatsd-host', default=settings.DOGSTATSD_HOST, help='Host to send DogStatsD metrics to.')
 @click.option('--dogstatsd-port', default=settings.DOGSTATSD_PORT, type=int, help='Port to send DogStatsD metrics to.')
-def replacer(replacements_topic, consumer_group, bootstrap_server, clickhouse_host, clickhouse_port, dataset,
+def replacer(replacements_topic, consumer_group, bootstrap_server, clickhouse_host, clickhouse_port, dataset_name: str,
              max_batch_size, max_batch_time_ms, auto_offset_reset, queued_max_messages_kbytes,
-             queued_min_messages, log_level, dogstatsd_host, dogstatsd_port):
+             queued_min_messages, log_level, dogstatsd_host, dogstatsd_port) -> None:
 
     import sentry_sdk
     from snuba import util
     from snuba.clickhouse.native import ClickhousePool
-    from batching_kafka_consumer import BatchingKafkaConsumer
-    from snuba.replacer import ReplacerWorker
+    from snuba.datasets.events import EventsDataset
+    from snuba.replacer import Replacement, ReplacerWorker
+    from snuba.utils.kafka.consumers.batching import BatchingKafkaConsumer
 
     sentry_sdk.init(dsn=settings.SENTRY_DSN)
-    dataset = get_dataset(dataset)
+    dataset = get_dataset(dataset_name)
+    if not isinstance(dataset, EventsDataset):
+        raise TypeError('replacer only compatible with EventsDataset')
```

Comment:
This is required since the `ReplacerWorker` only accepts `EventsDataset` (see description.)